huaweicloud/services/iotda/resource_huaweicloud_iotda_product.go

```diff
@@ -0,0 +1,604 @@
+										),
+									},
+
+									"paras": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Computed: true,
+										MaxItems: 500,
+										Elem:     propertySchema("services.commands.paras"),
+									},
+
+									"responses": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Computed: true,
+										MaxItems: 500,
+										Elem:     propertySchema("services.commands.responses"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"manufacturer_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 32),
+					validation.StringMatch(regexp.MustCompile(stringRegxp), stringFormatMsg),
+				),
+			},
+
+			"industry": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 64),
+					validation.StringMatch(regexp.MustCompile(stringRegxp), stringFormatMsg),
+				),
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 128),
+					validation.StringMatch(regexp.MustCompile(stringRegxp), stringFormatMsg),
+				),
+			},
+
+			"space": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"product_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 32),
+					validation.StringMatch(regexp.MustCompile(`^[A-Za-z-_0-9]+?`),
+						"Only letters, digits, underscores (_) and hyphens (-) are allowed."),
+				),
+			},
+		},
+	}
+}
+
+// propertySchema get the schema define for services.properties; services.commands.paras; services.commands.responses
+func propertySchema(category string) *schema.Resource {
+	common := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
+					validation.StringMatch(regexp.MustCompile(stringRegxp), stringFormatMsg),
+				),
+			},
+
+			"type": { // keep same with console
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{"int", "decimal", "string", "DateTime",
+					"jsonObject", "string list"}, false),
+			},
+
+			"enum_list": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"min": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(-2147483647, 2147483647),
+			},
+
+			"max": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      65535,
```

Comment:
The default value is this on Console side and should not be empty, I'd like to know what it will be after created if we don't have this Default.

Reply:
the default value of max is 0.